### ISSUE TITLE
Merge the login and signup screens

### ISSUE THREAD

**Issue by jidanni** [openstreetmap/openstreetmap-website#1567]
User visits https://www.openstreetmap.org/login and sees
```
Alternatively, use a third party to login:
```
OK, he clicks one of them and sees
```
Your ID is not associated with a OpenStreetMap account yet.

If you are new to OpenStreetMap, please create a new account using the form below.

If you already have an account, ADD:***DO NOT USE THE FORM BELOW!!!*** you can login to your account using your
username and password and then associate the account with your ID in
your user settings.
```
Be sure to add the "DO NOT USE THE FORM BELOW" where I put it, else the user very likely will just fill in the boxes, creating a superfluous account, etc.

**Comment by tomhughes** [comment]
No that is a completely ridiculous suggestion.

I agree there's a problem, but the solution is to merge the login and signup screens not to add silly warnings.

**Comment by jidanni** [comment]
How are you going to remember to merge them without any open bug related to it?

**Comment by tomhughes** [comment]
Why should I need to remember it? I think it's a good idea but I have no immediate plans to do it myself so I'm not sure why I need to remember it? 

If you're suggesting this ticket should stand as an "idea" for random passers by then it clearly doesn't do that as it stands, and it might be better to enter a new one rather than try and clean this up. That said I'm not a great fan of assembling long to-do lists in the ticketing system in the absence of volunteers to work on them.

**Comment by jidanni** [comment]
OK I retitled it. But I cannot reopen it. Perhaps somebody else can. That way one day when there is a volunteer, the problem will get fixed. Else it will not get fixed. Thank you.

**Comment by tomhughes** [comment]
The problem is the useful information is buried well down the ticket and the initial description is now totally misleading. That said I will reopen it, though I will make the description a bit more general.

**Comment by tomhughes** [comment]
It occurs to me that one quick fix that would be much easier than merging the screens would be adding the social login buttons to the signup page.

**Comment by Zverik** [comment]
But doesn't the login page already have social login buttons?
I think it's the other way around: registration page should have an option to login with existing account.

**Comment by tomhughes** [comment]
@Zverik Yes the login page already has them but I suggested adding them to the signup page so I'm not sure what you're getting at.

@jidanni Please stop changing the subject - it was set to something that attempted to describe the problem we are aiming to solve and you have changed it to mention one possible solution.

**Comment by Zverik** [comment]
Ah, I misread "signup" as "signin", sorry. But to me it doesn't seem like the issue from the first post.

As I understand, this issue is linked with #1274: when you have an account in OSM already, after clicking on a social login button you should be notified that instead of creating a new account, you can link that to your existing account.

**Comment by tomhughes** [comment]
Ah that's a different problem, but I don't see how it would work because how would we know they already had an account?

I mean sometime there might be a matching email address but certainly not always.
